fluss-server/src/main/java/org/apache/fluss/server/coordinator/event/watcher/TableChangeWatcher.java

```diff
@@ -231,4 +242,31 @@ private void processCreatePartition(
                             tablePath, tableId, partitionId, partitionName, partitionAssignment));
         }
     }
+
+    private void processSchemaChange(TablePath tablePath, int schemaId) {
+
+        try {
+            int currentSchemaId = zooKeeperClient.getCurrentSchemaId(tablePath);
+            SchemaInfo schemaInfo;
+            if (schemaId != currentSchemaId) {
+                LOG.warn(
+                        "Schema id {} is not equal to current schema id {}. Skipping schema change processing.",
+                        schemaId,
+                        currentSchemaId);
+                return;
+            }
```

Comment:
Why check current schema id? This is a heavy operation. Even if this is an old schema, I think it is still fine to process the schema.